### DISCUSSION TITLE
OPS Package Troubleshooting

### DISCUSSION THREAD

**Discussion by XanderCrews66** [nanovms/ops#1351] [Q&A] [answered]
I am attempting to ```ops pkg load``` the bind package that is hosted on repo.ops.city. From an OPS perspective it works fine, I can see OPS pull the package and start named with the arguments supplied in the `package.manifest`, but named exits with the following error:

`fflush() to pid file '/var/run/named/named.pid' failed` 

I can run this binary locally (`~/.ops/packages/eyberg/bind_9.16.4/named -4 -p 8081 -f -g`) and it starts fine.

My dilemma is I can't tell what is supposed to be an OPS/nanos problem or a binary specific problem. In a situation like this, where the binary starts through OPS but has an error, what is the procedure? I have a feeling BIND 9 would tell me to make an issue here, and you guys could tell me to create an issue with BIND 9. 

Thanks for any input

**Comment by eyberg** [comment, marked as answer]
this is a fairly old package that should probably be updated with a base volume size set as a default - it is failing to flush cause it ran out of space - if you set one via config it loads for me ->

```
ops pkg load eyberg/bind:9.13.4 -p 8081 -c config.json
..... *snip* ....
15-Jul-2022 21:20:38.771 zone_settimer: zone hostname.bind/CH: enter
15-Jul-2022 21:20:38.772 managed-keys-zone: Key 20326 for zone . is now trusted (initializing key verified)
15-Jul-2022 21:20:38.775 set_refreshkeytimer: managed-keys-zone: enter
15-Jul-2022 21:20:38.776 managed-keys-zone: next key refresh: 16-Jul-2022 21:20:38.776
15-Jul-2022 21:20:38.778 zone_settimer: managed-keys-zone: enter
15-Jul-2022 21:20:38.779 zone_journal: managed-keys-zone: enter
15-Jul-2022 21:20:38.783 zone_needdump: managed-keys-zone: enter
15-Jul-2022 21:20:38.784 zone_settimer: managed-keys-zone: enter
15-Jul-2022 21:20:38.786 resolver priming query complete
en1: assigned FE80::F0E0:4CFF:FEA0:B604
```

```
➜  ~ cat config.json
{
    "BaseVolumeSz": "100m"
}
```

**Reply by XanderCrews66** [reply]
Makes sense, I added the volume and both this package and a local package using the current build for named worked fine. When you say to add a base volume size set as a default, Is that something I can put in the manifest? 

```
$ cat package.manifest
{
    "Program":"named_9.19.4-dev/named",
    "Args":["named", "-4", "-f", "-g"],
    "Version":"9.19.4-dev", 
    "BaseVolumeSz": "100m"
}
```

**Reply by eyberg** [reply]
yes - if you are just playing around locally you can pass it an extra config.json and that will apply on top of whatever is in the package.manifest; you can also create a new package with 'BaseVolumeSz' set and that way you don't have to apply a secondary config to it - the BaseVolumeSz only takes affect upon image creation so if you add it to a new package it doesn't bump the package size up by 100mb
